### DISCUSSION TITLE
Question about "Partial Data Acceptance"

### DISCUSSION THREAD

**Discussion by Wizmann** [microsoft/msquic#2240] [Q&A] [answered]
Could you please give more details about the differences between returning `QUIC_STATUS_SUCCESS` and `QUIC_STATUS_CONTINUE` in "partial data acceptance" sceanrio?

It said [that][1] by returning `QUIC_STATUS_CONTINUE` the app could "immediately get a callback with the rest of the data". And when will the app get the callback by returning `QUIC_STATUS_SUCCESS`? Will it wait for the next coming packet(s) or wait for a specfic time period?

[1]: https://github.com/microsoft/msquic/blob/main/docs/Streams.md#partial-data-acceptance

**Comment by nibanks** [comment, marked as answer]
If you consume the whole buffer, there is no difference between `QUIC_STATUS_SUCCESS` and `QUIC_STATUS_CONTINUE`. If you partially consume the buffer and return `QUIC_STATUS_SUCCESS` then future receive callbacks are immediately paused. If you return `QUIC_STATUS_CONTINUE` the callbacks are not paused. The rest of the data you didn't consume will immediately be indicated in a new receive callback.

**Reply by Wizmann** [reply]
Is it costly to call `StreamReceiveSetEnabled`? And will it immediately trigger the callback if we (re)enable the stream receive?

**Reply by nibanks** [reply]
`StreamReceiveSetEnabled` will enable/unpause the receive callbacks. It is an async call, so it's fairly cheap, but as with all things performance related, it's relative, and you will have to try things out and measure it to see how costly it is in your scenarios.
